### PR TITLE
Promotion fix

### PR DESCRIPTION
**Pull request by Idonotus**
Fixes an oversight where piece that can be promoted to but where not on the board did not have correct slide lines

### CODE REVIEW

File: src/client/scripts/game/chess/variant.js [hunk 1]
```diff
@@ -55,10 +55,16 @@ const variant = (function() {
      */
     function initExistingTypes(gamefile) {
         const teamtypes = new Set(Object.values(gamefile.startSnapshot.position)); // Make a set of all pieces in game
```
Comment (Naviary3) [inline src/client/scripts/game/chess/variant.js:57]: Can this make use of gamefile.startSnapshot.existingTypes? Or is that not defined yet by this point?

**Comment by Naviary3** [comment]
It's worth asking if gamefile.startSnapshot.existingTypes should also be update to include all possible promotions. What instances in the code is it used in? And does it also need to include possible promotions?

**Comment by Idonotus** [comment]
This change already adds promotions to `existingTypes`. This function is responsible for init `existingTypes`. Is that what you were asking? It seems to only be used for `slidingPossible` but i think with some changes it ***could*** be used to optimize adjacency checks

**Comment by Naviary3** [comment]
> This function is responsible for init `existingTypes`

Oh right, of course, shoulda read closer, I think I thought it was responsible for calculating slidingPossible

**Comment by Naviary3** [comment]
Thanks!

**Pull request merged**